orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/RunTaskHandler.kt

```diff
@@ -189,13 +190,15 @@ class RunTaskHandler(
                   log.error("Error running ${message.taskType.simpleName} for ${message.executionType}[${message.executionId}]", e)
                 }
               }
-
+              taskException = e;
```

Comment:
is there a reason not to set taskException for the other cases in the catch block ?

Reply:
My original thought was that those other blocks retry, but thinking about it more no, it should be set right away.

I'll move it up.